### PR TITLE
Fix invalid access of weakly owned ptr in iOS a11y bridge

### PR DESCRIPTION
**Pull request by dnfield**
This fixes the crash part of https://github.com/flutter/flutter/issues/92853

Does not fix the focus getting trapped.

/cc @chunhtai fyi

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm [hunk 1]
```diff
@@ -6,13 +6,11 @@
...
 #import "flutter/shell/platform/darwin/ios/framework/Source/SemanticsObject.h"
 
-@interface FlutterSemanticsScrollView ()
-@property(nonatomic, assign) SemanticsObject* semanticsObject;
-@end
-
-@implementation FlutterSemanticsScrollView
+@implementation FlutterSemanticsScrollView {
+  fml::WeakPtr<SemanticsObject> _semanticsObject;
+}
```
Comment (jmagman) [inline shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm:11]: Is there something special about `WeakPtr`, or does it work if you keep the Objective-C property and just make it weak?
```suggestion
@interface FlutterSemanticsScrollView ()
@property(nonatomic, weak) SemanticsObject* semanticsObject;
@end
```

File: shell/platform/darwin/ios/BUILD.gn [hunk 2]
```diff
@@ -256,6 +256,7 @@ shared_library("ios_test_flutter") {
     "framework/Source/connection_collection_test.mm",
   ]
   deps = [
+    ":flutter_framework",
```
Comment (jmagman) [inline shell/platform/darwin/ios/BUILD.gn:259]: What was this change for? I think the tests currently import individual headers instead of the framework.

Comment (dnfield) [reply shell/platform/darwin/ios/BUILD.gn:259]: The target depends on this being built to run, but isn't declaring it as a dep. I found this out by building the target before building this dep.

File: shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm (hunk 1 above)
Comment (dnfield) [reply shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm:11]: You can't use weak in MRC right?

Comment (jmagman) [reply shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm:11]: Argh right, it's already `assign`.  And I can't remember how to get this pattern to not `message sent to deallocated instance`.

**Review by jmagman — APPROVED** [review]
LGTM, thanks @dnfield

File: shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm (hunk 1 above)
Comment (jmagman) [reply shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm:11]: Oh, I remember, `FlutterScrollableSemanticsObject` would have to set the scrollView's semanticsObject to `nil` in dealloc.

Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm:11]: Yea, you only get auto nilling variables with ARC.

Who is the real owner of the semanticsObject?  Do they have a reference to the FlutterSemanticsScrollView?  They can nil out the property.

Comment (jmagman) [reply shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm:11]: > Oh, I remember, `FlutterScrollableSemanticsObject` would have to set the scrollView's semanticsObject to `nil` in dealloc.

https://github.com/flutter/engine/blob/48ce284dff9cd876486542b0c8df850c9ffd5903/shell/platform/darwin/ios/framework/Source/SemanticsObject.mm#L168-L170
like
https://github.com/flutter/engine/blob/48ce284dff9cd876486542b0c8df850c9ffd5903/shell/platform/darwin/ios/framework/Source/SemanticsObject.mm#L301

Comment (dnfield) [reply shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm:11]: I tried something close to this and couldn't get it working.  Should I try again or just keep with the weak factory?

File: shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.h [hunk 3]
```diff
@@ -20,6 +20,8 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @interface FlutterSemanticsScrollView : UIScrollView
 
+@property(nonatomic, assign) SemanticsObject* _Nullable semanticsObject;
```
Comment (jmagman) [inline shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.h:23]:
```suggestion
@property(nonatomic, assign, nullable) SemanticsObject* semanticsObject;
```

**Review by jmagman — APPROVED** [review]
LGTM with `nullable` nit.

File: shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.h (hunk 3 above)
Comment (dnfield) [reply shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.h]: Live footage of me writing ObjC code:
https://memegenerator.net/img/instances/72953023/i-have-no-idea-what-im-doing.jpg

**Comment by dnfield** [comment]
Hm. @jmagman - for some reason, this version still crashes (evne though the unit test passes).

**Comment by dnfield** [comment]
Sorry, I was using the wrong binary. It works fine.

**Pull request merged**